src/pytorch_lightning/core/module.py

```diff
@@ -177,13 +182,15 @@ def lr_schedulers(self) -> Optional[Union[LRSchedulerTypeUnion, List[LRScheduler
     def trainer(self) -> "pl.Trainer":
         if not self._running_torchscript and self._trainer is None:
             raise RuntimeError(f"{self.__class__.__qualname__} is not attached to a `Trainer`.")
-        return self._trainer
+        if self._trainer is not None:
+            return self._trainer
```

Comment:
I have to check locally however I'm pretty certain I did this to avoid a return-value error

Reply:
@carmocca added this.

what's up with `_running_torchscript`? if it's True and trainer is None, should it return None or raise an error?

Reply:
The running torchscript hack is required by a bug in PyTorch's jit (https://github.com/pytorch/pytorch/issues/67146), which makes `mypy` get confused. But this would never return `None` normally.

My suggestion is to undo this change and add an ignore for this error. Hopefully they fix this bug eventually.